Blurringoperator

jakobsj opened the pull request:
Adds new BlurringOperator to address #320. Seems to work. Not sure whether adjoint is correct, but dot test seems happy.

jakobsj commented:
I updated this from "draft PR" to "ready for review PR", if you can have a look. Builds locally.

jakobsj commented:
Closes #320.

epapoutsellis commented:
Fine with me, only some minor changes:

1) Check PSF numpy type in the init.
2) This will work only with 2D cases, so I suggest to name it BlurringOperator2D. Probably we can make it work for other dimensions but it will be slow anyway. So it better to start working on FFT
3) Move demo in CIL-demos issues.

jakobsj commented:
Thanks for looking at it.

1. What exactly should be checked in init?
2. It was indeed written for the 2D case, but since it is using scipy.ndimage I believe it will actually work for any dimension, but need to test.
3. Yes, will move the demo. Not sure if we want to keep it here in the main as well?

Also, I need to remove line 42, since it is not being used when using correlate for the adjoint.

epapoutsellis commented:
1. if PSF not numpy raise an error. because usually everything we do is with datacontainters.
2. i think you are right. If it works for small 3D, then current  name is fine

jakobsj commented:
@epapoutsellis @paskino   I have updated to address Vaggelis' comments to do input type check and added the demo to CIL-demos as PR https://github.com/vais-ral/CIL-Demos/pull/47, please can you have a look if ready to merge, thanks!

epapoutsellis commented:
Fine with me. One small addition. Could you add a small test in test directory checking this [line](https://github.com/vais-ral/CCPi-Framework/pull/565/files#diff-314a9c91744ff6714c6c8d320c1353ccR130)

jakobsj commented:
OK, hadn't seen the request to add a unit test. That is a good idea, e185038 adds that. @paskino OK to merge?
